### ISSUE TITLE
questions on return-generation-logits when streaming

### ISSUE THREAD

**Issue by binhtranmcs** [NVIDIA/TensorRT-LLM#2090]
I see that when run streaming inference, the [result](https://github.com/NVIDIA/TensorRT-LLM/blob/main/cpp/include/tensorrt_llm/executor/executor.h#L323) contains `generationLogits` for the full sequence. Which means it will contain a tensor of shape `batch_size*beam_size*max_output_length*vocab_size` every time the `executor` returns a token, which is very inefficient. Is this expected behavior? I believe returning only the logits for the current token would be more optimal.

Please have a look. Thanks in advance!

**Comment by AdamzNV** [comment]
It's under our consideration now.
